### PR TITLE
Isla 2028 accessforsetting

### PR DESCRIPTION
**Pull request by sd-cyberinc**

### CODE REVIEW

File: settingsmetadata/model/SettingsMetaData.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	DisplaySequence int     `json:"displaySequence"`
+	Type            string  `json:"type"`
+	TypeParam       string  `json:"typeParam"`
+	Default         string  `json:"default"`
+	Required        bool    `json:"required"`
+	Validation      string  `json:"validation"`
+	MaxValue        float32 `json:"maxValue"`
+	MinValue        float32 `json:"minValue"`
+	Hidden          bool    `json:"hidden"` //remove this function before sending Merge request
+	//Access          string  `json:"access"` //remove this function before sending Merge request
```
Comment (dhavalgogri17) [inline settingsmetadata/model/SettingsMetaData.go:26]: Can you remove the commented code

File: settingsmetadata/model/SettingsMetaData.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+			return
+		}
+	}
+	return
+}
+
+/*
+
+
+func (metadata *SettingsMetaData) ParseAndValidate(value interface{}) (interface{}, error) {
```
Comment (dhavalgogri17) [inline settingsmetadata/model/SettingsMetaData.go:42]: Can you remove the commented code

File: settingsmetadata/repository/tenantSettingsRepository.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	returnMap := make(map[string]string)
+	for key, setting := range settingsMap {
+		returnMap[key] = fmt.Sprintf("%v", setting)
+	}
+
+	return returnMap, nil
+}
+
+func (tenantRepository *gormTenantSettingsRepository) checkAndInitializeSettingsMetadata() error {
+	if len(tenantRepository.settingsMetadatas) == 0 {
```
Comment (dhavalgogri17) [inline settingsmetadata/repository/tenantSettingsRepository.go:72]: controller.app.Config.IsSet(config.EvSuffixForSettingsMetadataPath) check is not present here the way it has been done in settingmetadata controller

File: settingsmetadata/repository/tenantSettingsRepository.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+		settingMetadata, err := tenantRepository.initSettingsMetaData(config.EvSuffixForSettingsMetadataPath)
+		if err != nil {
+			return err
+		}
+		tenantRepository.settingsMetadatas = settingMetadata
+	}
+	if len(tenantRepository.globalsettingsMetadatas) == 0 {
```
Comment (dhavalgogri17) [inline settingsmetadata/repository/tenantSettingsRepository.go:79]: controller.app.Config.IsSet(config.EvSuffixForGlobalSettingsMetadataPath) is not present here the way it has been done in settingsmetadata controller

File: settingsmetadata/controllers/SettingsMetadataController.go [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+
+	settings, err := tenant.GetSettings()
+	if err != nil {
+		context.LogError(err, microappLog.MessageGetEntityError)
+		microappWeb.RespondError(w, err)
+		return
+	}
+
+	/*
+		for key, settingsValue := range settings {
```
Comment (dhavalgogri17) [inline settingsmetadata/controllers/SettingsMetadataController.go:214]: remove commented code if no use

File: settingsmetadata/model/SettingsMetaData.go [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+	errors := make(map[string]string)
```
Comment (dhavalgogri17) [inline settingsmetadata/model/SettingsMetaData.go:43]: remove commented code if no use

File: settingsmetadata/repository/tenantSettingsRepository.go [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+	//convert to map[string]string
+	returnMap := make(map[string]string)
+	for key, setting := range settingsMap {
+		returnMap[key] = fmt.Sprintf("%v", setting)
+	}
+
+	return returnMap, nil
+}
+
+func (tenantRepository *gormTenantSettingsRepository) checkAndInitializeSettingsMetadata() error {
```
Comment (dhavalgogri17) [inline settingsmetadata/repository/tenantSettingsRepository.go:71]: The method is different from teh one used in settingsmetadata. Additional check of controller.app.Config.IsSet(config.EvSuffixForSettingsMetadataPath) and controller.app.Config.IsSet(config.EvSuffixForGlobalSettingsMetadataPath) is not present in if condition.

**Review by dhavalgogri17 — CHANGES_REQUESTED** [review]

File: settingsmetadata/model/SettingsMetaData.go (hunk 1 above)
Comment (sd-cyberinc) [reply settingsmetadata/model/SettingsMetaData.go]: removed

File: settingsmetadata/repository/tenantSettingsRepository.go (hunk 3 above)
Comment (sd-cyberinc) [reply settingsmetadata/repository/tenantSettingsRepository.go]: removed

File: settingsmetadata/controllers/SettingsMetadataController.go (hunk 5 above)
Comment (sd-cyberinc) [reply settingsmetadata/controllers/SettingsMetadataController.go]: removed

File: settingsmetadata/repository/tenantSettingsRepository.go (hunk 4 above)
Comment (sd-cyberinc) [reply settingsmetadata/repository/tenantSettingsRepository.go]: condition added

File: settingsmetadata/model/SettingsMetaData.go (hunk 6 above)
Comment (sd-cyberinc) [reply settingsmetadata/model/SettingsMetaData.go]: removed

File: settingsmetadata/repository/tenantSettingsRepository.go (hunk 7 above)
Comment (sd-cyberinc) [reply settingsmetadata/repository/tenantSettingsRepository.go:71]: condition added

File: settingsmetadata/controllers/SettingsMetadataController.go [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+	if tenant.Settings != "{}" {
+		queryProcessor := []repository.QueryProcessor{repository.Filter("id = ?", tenantID)}
+		err = controller.repository.Upsert(uow, &tenant, queryProcessor)
+		if err != nil {
+			context.LogError(err, microappLog.MessageUpdateEntityError)
+			microappWeb.RespondError(w, err)
+			return
+		}
+	} else {
+		err = controller.repository.Delete(uow, tenantModel.TenantSettings{}, tenantID)
```
Comment (dhavalgogri17) [inline settingsmetadata/controllers/SettingsMetadataController.go:171]: This has to be a permanent delete instead of a soft delete

**Review by dhavalgogri17 — CHANGES_REQUESTED** [review]

File: settingsmetadata/controllers/SettingsMetadataController.go (hunk 8 above)
Comment (sd-cyberinc) [reply settingsmetadata/controllers/SettingsMetadataController.go]: changed to DeletePermenant

**Review by dhavalgogri17 — APPROVED** [review]

**Pull request merged**